Xamarin.Forms.Core/Markup/BindableObjectExtensions.cs

```diff
@@ -0,0 +1,183 @@
+			object targetNullValue = null,
+			object fallbackValue = null
+		) where TBindable : BindableObject
+		{
+			bindable.Bind(
+				DefaultBindableProperties.GetFor(bindable),
+				path, mode, converter, converterParameter, stringFormat, source, targetNullValue, fallbackValue
+			);
+			return bindable;
+		}
+
+		/// <summary>Bind to the default property with inline conversion</summary>
+		public static TBindable Bind<TBindable, TSource, TDest>(
+			this TBindable bindable,
+			string path = bindingContextPath,
+			BindingMode mode = BindingMode.Default,
+			Func<TSource, TDest> convert = null,
+			Func<TDest, TSource> convertBack = null,
+			object converterParameter = null,
+			string stringFormat = null,
+			object source = null,
+			object targetNullValue = null,
+			object fallbackValue = null
+		) where TBindable : BindableObject
+		{
+			var converter = new FuncConverter<TSource, TDest, object>(convert, convertBack);
+			bindable.Bind(
+				DefaultBindableProperties.GetFor(bindable),
+				path, mode, converter, converterParameter, stringFormat, source, targetNullValue, fallbackValue
+			);
+			return bindable;
+		}
+
+		/// <summary>Bind to the default property with inline conversion and conversion parameter</summary>
+		public static TBindable Bind<TBindable, TSource, TParam, TDest>(
+			this TBindable bindable,
+			string path = bindingContextPath,
+			BindingMode mode = BindingMode.Default,
+			Func<TSource, TParam, TDest> convert = null,
+			Func<TDest, TParam, TSource> convertBack = null,
+			object converterParameter = null,
+			string stringFormat = null,
+			object source = null,
+			object targetNullValue = null,
+			object fallbackValue = null
+		) where TBindable : BindableObject
+		{
+			var converter = new FuncConverter<TSource, TDest, TParam>(convert, convertBack);
+			bindable.Bind(
+				DefaultBindableProperties.GetFor(bindable),
+				path, mode, converter, converterParameter, stringFormat, source, targetNullValue, fallbackValue
+			);
+			return bindable;
+		}
+
+		/// <summary>Bind to the <typeparamref name="TBindable"/>'s default Command and CommandParameter properties </summary>
+		/// <param name="parameterPath">If null, no binding is created for the CommandParameter property</param>
+		public static TBindable BindCommand<TBindable>(
+			this TBindable bindable,
+
+			string path = bindingContextPath,
+			object source = null,
+			string parameterPath = bindingContextPath,
+			object parameterSource = null
+		) where TBindable : BindableObject
+		{
+			(var commandProperty, var parameterProperty) = DefaultBindableProperties.GetForCommand(bindable);
+
+			bindable.SetBinding(commandProperty, new Binding(path: path, source: source));
+
+			if (parameterPath != null)
+				bindable.SetBinding(parameterProperty, new Binding(path: parameterPath, source: parameterSource));
+
+			return bindable;
+		}
+
+		public static TBindable Assign<TBindable, TVariable>(this TBindable bindable, out TVariable variable)
```

Comment:
I've doubts about this

Reply:
The purpose of `Assign` and `Invoke` is to promote clean separation of UI markup from UI logic.
Using partial classes, you can then seperate them in two files, e.g.:

`RegistrationCodePage.cs`:
```CSharp
new Entry { }
    .Assign (out registrationCodeEntry),

new ListView { }
    .Invoke (list => list.ItemSelected += List_ItemSelected),
```
`RegistrationCodePage.logic.cs`:
```CSharp
Entry registrationCodeEntry;

protected override void HandleOnAppearing()
{
    base.HandleOnAppearing();
    registrationCodeEntry?.Unfocus();
}

void List_ItemSelected(object sender, SelectedItemChangedEventArgs e) { ... }
```
If this does not resolve your doubts, could you expand a bit?